CONTRIBUTING.md

```diff
@@ -0,0 +1,166 @@
+- Avoid to speak about keys and other hard to understand things in the primary UI.
+- The app shall work offline as well as with bad network.
+- Users do not read (much).
+- Consistency matters.
+- Offer only things that are highly useful to many people in primary UI.
+  If really needed, bury other things eg. in some menus.
+- The app should be for the many, not for the few.
+
+
+## Contributing Code
+
+The [README](./README.md) explains how to set up the build environment.
+If you run into troubles,
+ask on one of the [communication channels](https://delta.chat/contribute) for help.
+
+To contribute code,
+[open a Pull Request](https://github.com/deltachat/deltachat-ios/pulls).
+
+If you have write access to the repository,
+push a branch named `<username>/<feature>`
+so it is clear who is responsible for the branch,
+and open a PR proposing to merge the change.
+Otherwise fork the repository and create a branch in your fork.
+
+Please add a meaningful description to your PR
+so that reviewers get an idea about what the modifications are supposed to do.
+
+A meaningful PR title is helpful for [updating `CHANGELOG.md` on releases](./RELEASE.md)
+(CHANGELOG.md is updated manually
+to only add things that are at least roughly understandable by the end user)
+
+If the changes affect the user interface,
+screenshots are very helpful,
+esp. before/after screenshots.
+
+
+### Coding Conventions
+
+Source files are partly derived from different other open source projects
+and may follow different coding styles and conventions.
+
+If you do a PR fixing a bug or adding a feature,
+please embrace the coding convention you see in the corresponding files,
+so that the result fits well together.
+
+Do not refactor or rename things in the same PR
+to make the diff small and the PR easy to review.
+
+Project language is Swift.
+To support a wider range of devices, SwiftUI is not used currently.
+This also avoids conditional code and allows SwiftUI to mature meanwhile.
+
+By using [Delta Chat Core](https://github.com/deltachat/deltachat-core-rust)
+there is already a strong separation between "UI" and "Model".
+Further separations and abstraction layers are often not helpful
+and only add more complexity.
+Eg. Model-View-ViewModel (MVVM), often hailed as best practise,
+is easily overkill in case of deltachat-ios.
+In existing code, functionality is just added to the ViewControllers,
+which is easy to understand and good enough.
+
+Try to avoid premature optimisation
+and complexity because it "may be needed in some future".
+Usually, it is not.
+
+Note, that most current iOS developers are not "iOS natives".
+
+Readable code is better than having some paradigms fulfilled.
+Keep things really on point and simple.
+If this gets in conflict with embracing existing style, however,
+consistency with existing code is more important.
+
+Do as many things as possible in an "iOS way" with APIs the system provides
+and avoid using external libraries;
```

Comment:
also many of the available swift libraries don't have the highest quality and some even come with their own bugs and build problems.